Rewrite HTTP OAuth2 client BBEs (#3793)
ballerina-platform/ballerina-distribution: examples/http-client-oauth2-refresh-token-grant-type/http_client_oauth2_refresh_token_grant_type.md

# HTTP client - OAuth2 refresh token grant type

> The `http:Client` can be secured with OAuth2 refresh token grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration. Use this to automatically retrieve an access token when it is expired.

Author: ```suggestion
The `http:Client` can connect to a service that is secured with the OAuth2 refresh token grant type by adding the `Authorization: Bearer <token>` header to each request. The required configurations for this grant type can be specified in the `auth` field of the client configuration. Use this to automatically retrieve an access token when it is expired.
```

> An `http:Client` can be secured with OAuth2 refresh token grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 refresh token grant type can be specified in the `auth` field as the `http:OAuth2RefreshTokenGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.

Reviewer: **When to use ?**
use when client need to  exchange a refresh token for an access token when the access token has expired. This allows clients to continue to have a valid access token without further interaction with the user

Author: Added via [187d99d](https://github.com/ballerina-platform/ballerina-distribution/pull/3793/commits/187d99dffcd52f92e019e05f224ac4dc5e4a2e66)

> An `http:Client` can be secured with OAuth2 refresh token grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 refresh token grant type can be specified in the `auth` field as the `http:OAuth2RefreshTokenGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.

Reviewer 2: ```suggestion
The `http:Client` can be secured with OAuth2 refresh token grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 refresh token grant type can be specified in the `auth` field as the `http:OAuth2RefreshTokenGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.
```

> The `http:Client` can be secured with OAuth2 refresh token grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 refresh token grant type can be specified in the `auth` field as the `http:OAuth2RefreshTokenGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.

Author: ```suggestion
The `http:Client` can be secured with OAuth2 refresh token grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 refresh token grant type can be specified in the `auth` field as the `http:OAuth2RefreshTokenGrantConfig` record. Use this grant type when the client needs to exchange a refresh token for an access token if the access token has expired. This allows the client to have a valid access token without further interaction with the user.
```

> The `http:Client` can be secured with OAuth2 refresh token grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration. Use this grant type when the client needs to exchange a refresh token for an access token if the access token has expired. This allows the client to have a valid access token without further interaction with the user.

Author: ```suggestion
The `http:Client` can be secured with OAuth2 refresh token grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration. Use this to automatically retrieve an access token when it is expired.
```

> The `http:Client` can be secured with OAuth2 refresh token grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration. Use this grant type when the client needs to exchange a refresh token for an access token if the access token has expired. This allows the client to have a valid access token without further interaction with the user.

Author: Fixed via https://github.com/ballerina-platform/ballerina-distribution/pull/3793/commits/373fd382554d45179c4ec44e8ea2280b4b8709b2

> The `http:Client` can be secured with OAuth2 refresh token grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 refresh token grant type can be specified in the `auth` field as the `http:OAuth2RefreshTokenGrantConfig` record. Use this grant type when the client needs to exchange a refresh token for an access token if the access token has expired. This allows the client to have a valid access token without further interaction with the user.

Author: ```suggestion
The `http:Client` can be secured with OAuth2 refresh token grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration. Use this grant type when the client needs to exchange a refresh token for an access token if the access token has expired. This allows the client to have a valid access token without further interaction with the user.
```
